### ISSUE TITLE
Refresh fails if aws loggroup is manually deleted

### ISSUE THREAD

**Issue by foragerr** [terraform-providers/terraform-provider-aws#1405]
### Terraform Version
0.9.2
0.9.9

previously reported for 

0.8.5
0.7.3

### Affected Resource(s)
aws_cloudwatch_log_subscription_filter
aws_cloudwatch_log_group

### Terraform Configuration Files
```hcl
resource "aws_cloudwatch_log_group" "lambda" {
    name = "/aws/lambda/${aws_lambda_function.lambda_function.function_name}"
    retention_in_days = "7"
}

resource "aws_cloudwatch_log_subscription_filter" "kinesis" {
    depends_on = [ "aws_cloudwatch_log_group.lambda" ]
    name = "my-kinesis-subscription"
    log_group_name = "/aws/lambda/${aws_lambda_function.lambda_function.function_name}"
    filter_pattern = ""
    destination_arn = "arn:aws:kinesis:ap-southeast-2:xxxxxxxx:stream/cloudwatch-sumologic-stream-dev"
    role_arn = "arn:aws:iam::xxxxxxxxx:role/cloudwatch-sumo-role-dev"
}
```

### Debug Output
-

### Panic Output
-

### Expected Behavior
terraform detects that the subscription filter does not exist on refresh.

### Actual Behavior
terraform queries for a loggroup that does not exist anymore and receives an error message from aws. terraform fails to complete refresh.

### Steps to Reproduce
1. `terraform apply`, a loggroup is created and a subscription filter is attached to the log group.
2.  manually delete loggroup, outside terraform. 
3. On the next `terraform apply`, terraform correctly detects that the loggroup no longer exists. But throws an error and fails when attempting to refresh subscription filter.

### Important Factoids
terraform should not fail if the error on this line is that loggroup doesn't exist: https://github.com/hashicorp/terraform/blob/v0.9.2/builtin/providers/aws/resource_aws_cloudwatch_log_subscription_filter.go#L137

### References
This is the exact same issue reported here: https://github.com/hashicorp/terraform/issues/11750
however that issue was closed with a fix that does not address the reported issue. please see: https://github.com/hashicorp/terraform/issues/11750#issuecomment-320757651

**Comment by stack72** [comment]
Hi @foragerr 

I *finally* fixed this in #1414 - i even included an acceptance test to show the behaviour of when a LogGroup is removed (thus removing the subscription filter) then it will show a recreation is needed

Sorry it took so long to get back to you on

Paul

**Comment by foragerr** [comment]
Awesome, Thank you sir! @stack72
